omisego-sdk/src/main/java/co/omisego/omisego/websocket/channel/SocketChannel.kt

```diff
@@ -0,0 +1,186 @@
+package co.omisego.omisego.websocket.channel
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 3/5/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.model.socket.SocketSend
+import co.omisego.omisego.model.socket.SocketTopic
+import co.omisego.omisego.websocket.SocketChannelListener
+import co.omisego.omisego.websocket.SocketClientContract
+import co.omisego.omisego.websocket.SocketConnectionListener
+import co.omisego.omisego.websocket.SocketCustomEventListener
+import co.omisego.omisego.websocket.channel.SocketChannelContract.Dispatcher
+import co.omisego.omisego.websocket.channel.SocketChannelContract.SocketClient
+import co.omisego.omisego.websocket.channel.dispatcher.SocketDispatcherContract
+import co.omisego.omisego.websocket.enum.SocketEventSend
+import co.omisego.omisego.websocket.enum.SocketStatusCode
+import okhttp3.WebSocketListener
+
+/**
+ * A SocketChannel is responsible for handling join or leave from the socket channel.
+ *
+ * @param socketDispatcher A [Dispatcher] is responsible dispatch all events to the client.
+ * @param socketClient A [SocketClient] for sending a message to the eWallet web socket API and close the web socket connection.
+ */
+internal class SocketChannel(
+    override val socketDispatcher: SocketChannelContract.Dispatcher,
+    override val socketClient: SocketChannelContract.SocketClient
+) : SocketClientContract.Channel, SocketChannelContract.Channel, SocketDispatcherContract.SocketChannel {
+    private val channelSet: MutableSet<String> by lazy { mutableSetOf<String>() }
+
+    /**
+     * A [SocketMessageRef] is responsible for create unique ref value to be included in the [SocketSend] request.
+     */
+    override val socketMessageRef: SocketChannelContract.MessageRef = SocketMessageRef().apply { scheme = SocketMessageRef.SCHEME_JOIN }
```

Comment:
Not sure the `socket` prefix is needed in the name since we're already in the `SocketChannel` context :)